src/Service.Host/client/src/components/SerialNumbers.js

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState, useRef, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown, Loading, Title, SnackbarMessage } from '@linn-it/linn-form-components-library';
+import {
+    TextField,
+    InputAdornment,
+    Typography,
+    Table,
+    TableHead,
+    TableRow,
+    TableCell,
+    TableBody
+} from '@material-ui/core';
+import { getSernosNote } from '../selectors/sernosNotesSelectors';
+import SerialNumber from './SerialNumber';
+import Page from '../containers/Page';
+
+function SerialNumbers({
+    items,
+    loading,
+    fetchItems,
+    sernosNotes,
+    snackbarVisible,
+    sernosNoteLoading,
+    sernosNotesLoading,
+    addSernosNote,
+    updateSernosNote,
+    setSnackbarVisible
+}) {
+    const [searchTerm, setSearchTerm] = useState('');
+    const [salesArticles, setSalesArticles] = useState([]);
+    const [selectedArticle, setSelectedArticle] = useState('');
+    const [debounceTimer, setDebounceTimer] = useState(null);
+
+    const savedFetchItems = useRef();
+    const savedDebounceTimer = useRef();
+
+    useEffect(() => {
+        savedFetchItems.current = fetchItems;
+    }, [fetchItems]);
+
+    useEffect(() => {
+        savedDebounceTimer.current = debounceTimer;
+    }, [debounceTimer]);
+
```

Comment:
Using refs here so the search hook will only be run when the search term changes.